### ISSUE TITLE
[Dev Branch] - bitcoin-qt crash on Windows 10

### ISSUE THREAD

**Issue by Justaphf** [gandrewstone/bitcoin#915]
I've had the same crash 3 times now.  Running on Windows 10 with natively built binaries, via MinGW.  The circumstances are as follows.

Running 64-bit `bitcoin-qt.exe` built from 1b39d15 and subsequently dde2d83.  Application is just running in the background (tray icon only).  At some point, usually within 24 hours of starting, the application crashes after accepting a new block and updating the chain tip with a subsequent "socket select error".  The icon remains in the system tray until I mouse over it, then it disappears (standard Windows behavior for an application that was not shut down correctly).

Excerpts from the second two crash logs below (I didn't save the first log).

Crash 2 log:
```
2018-01-13 10:16:50 Acceptable block: ver:20000000 time:1515838574 size: 122338 Tx:237 Sig:808
2018-01-13 10:16:50 UpdateTip: new best=0000000000000000036998d9b0c0c19cd051da0c14112122ace740601e47b5b9  height=512731 bits=402900906 log2_work=87.032462  tx=246695482  date=2018-01-13 10:16:14 progress=0.999999  cache=2.1MiB(1465txo)
2018-01-13 10:19:47 socket select error The operation completed successfully.  (0)
```
Crash 3 log:
```
2018-01-15 01:43:57 Acceptable block: ver:20000000 time:1515980610 size: 7993366 Tx:1654 Sig:58483
2018-01-15 01:43:57 Leaving block file 991: CBlockFileInfo(blocks=23, size=128840705, heights=512961...512983, time=2018-01-14...2018-01-15)
2018-01-15 01:43:57 Pre-allocating up to position 0x1000000 in blk00992.dat
2018-01-15 01:43:58 Pre-allocating up to position 0x200000 in rev00992.dat
2018-01-15 01:43:58 UpdateTip: new best=00000000000000000137553dd4fbe021cd1c0b169bf41bbdcdfa7f8ff77a3e54  height=512984 bits=402852681 log2_work=87.03569  tx=246861926  date=2018-01-15 01:43:30 progress=0.999999  cache=33.1MiB(137447txo)
2018-01-15 01:45:21 socket select error The operation completed successfully.  (0)
```

The Windows event log seems to indicate that the crash is related to this "socket select error" as Windows logs application shutdown ~1 second after the last line is written to the log file.

Windows event log for crash 3 (timestamp: 2018-01-15 01:45:21 UTC):
```
Faulting application name: bitcoin-qt.exe, version: 1.2.0.0, time stamp: 0x05750150
Faulting module name: bitcoin-qt.exe, version: 1.2.0.0, time stamp: 0x05750150
Exception code: 0xc0000005
Fault offset: 0x00000000001330a3
Faulting process id: 0x26b4
Faulting application start time: 0x01d38d5a5c7fc429
Faulting application path: D:\Bitcoin\bitcoin-qt.exe
Faulting module path: D:\Bitcoin\bitcoin-qt.exe
Report Id: 34bbeafb-0f31-4dc5-978b-401da39d1864
Faulting package full name: 
Faulting package-relative application ID: 
```

I will rerun again with some additional debug logs turned on `net` and `thin` to see if this sheds more light on the issue.

**Comment by Justaphf** [comment]
This is corrected by #923 in dev.  #924 will backport the same fix to the BitcoinCore branch.

Will leave open until 924 is merged.

**Comment by Justaphf** [comment]
Fixes mentioned have been merged, closing.
